contracts/CurrencyNetwork.sol

```diff
@@ -147,6 +147,8 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
             !_preventMediatorInterests || (_preventMediatorInterests && _customInterests),
             "The prevent mediator interest strategy cannot be set without using custom interests."
         );
+        require(bytes(_name).length != 0, "Cannot have an empty name");
+        require(bytes(name).length == 0, "Currency Network already initialized");
```

Comment:
Please replace that with a public flag `isInitialized`. You have to look hard to spot the difference here.
The public flag might also be useful, otherwise they would also have to check name.lengh in order to determine if the contract is initialized.
Also move this check to the top of the function.